inc/stlink.h

```diff
@@ -120,7 +120,8 @@ enum stlink_flash_type {
     STLINK_FLASH_TYPE_UNKNOWN = 0,
     STLINK_FLASH_TYPE_F0,    // used by f0, f1 (except f1xl),f3. */
     STLINK_FLASH_TYPE_F1_XL, // f0 flash with dual bank, apparently */
-    STLINK_FLASH_TYPE_F4,    // used by f2, f4, f7 */
+    STLINK_FLASH_TYPE_F4,    // used by f2, f4 */
+    STLINK_FLASH_TYPE_F7,
```

Comment:
Does F7 really need some specific flash code ? I've not checked again but AFAIR, it was pretty similar to f2/f4 and could be treated as well (same reg base, same bits for base operation). That would avoid some copypasting.

Reply:
I couldn't find the OPTCR and OPTCR1 registers on the F4, so to distinguish I added a new flash type. If you have another suggestion how to do this, I'm open to it

Reply:
If possible, we should align with the official documentation here.

Reply:
What do you propose exactly?

Reply:
This is related to the previous discussion about `src/st-flash/flash.h`.
So if the enum value names don't match between the F7 and F4, and you are sure with it, we need to have specific code.
In this case this should resolve this thread as well.